api/v1/client/src/commonMain/kotlin/auth/AuthService.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.client.auth
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.forms.submitForm
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.Parameters
+import io.ktor.http.isSuccess
+
+/**
+ * A service to generate/refresh an OAuth token to authenticate with the ORT server.
+ */
+class AuthService(
+    private val client: HttpClient,
+    private val tokenUrl: String,
+    private val clientId: String
+) {
+    /**
+     * Generate a token for the given [username] and [password].
+     */
+    suspend fun generateToken(username: String, password: String): TokenInfo =
```

Comment:
I am not sure whether this service should be public and directly exposed to clients. I would rather expect that there is some `OrtServerClient` class (or similar) that acts as a facade for the ORT Server API. This class would handle authentication behind the scenes.

Reply:
There will be an `OrtServerClient` class in the upcoming PR that will  handle authentication in its factory function. It might be better to change the visibility to internal?

Reply:
Yes, this class should not be accessible from the outside.